[design] 산책 요청 수락 화면 구현

green-yoon87 opened the pull request:
### 🔖  Issue Number

close #66 

### 📙 작업 내역

> 산책 요청 화면을 구현했습니다. 
> 
- [x]  디자인에 맞게 산책 요청 화면 구현 
- [x] 스크롤 뷰 추가  
- [x] VIPER 패턴 적용 


https://github.com/user-attachments/assets/41062936-78ca-4104-a58b-2a89630a60b5


### 📝 PR 특이사항 

> PR을 볼 때 팀원에게 알려야 할 특이사항을 알려주세요.
> 
- 특이 사항 1
내용이 많아보이지만... VIPER 패턴이어서 많아 보입니다.... ㅎㅎ 
커밋 양이 많아서 제한 시간 표시와 제한시간 지난 후의 로직은 PR 분리해서 올리겠습니다.

green-yoon87 committed:
[fix] #66 LayoutConstant 속성 중 이름 수정 (edge를 설정하는 값)

green-yoon87 committed:
[feat] #66  content과 bound 사이의 padding이 상하좌우 존재하는 label 구현

green-yoon87 committed:
[feat] #66 프로필이 들어있는 ImageView 내부에 이름 label, keyword가 존재하는 뷰 구현

green-yoon87 committed:
[feat] #66 장소를 선택하는 뷰 구현

green-yoon87 committed:
[feat] #66  구현한 profileView, locationSelectionView 적용

green-yoon87 committed:
[feat] #66 산책 요청 응답 화면 레이아웃 구현

green-yoon87 committed:
[feat] #66 스크롤뷰 적용

green-yoon87 committed:
[feat] #66 산책요청 데이터 구조 선언

green-yoon87 committed:
[feat] #66 Entity 예시 데이터

뷰 작업하면서 예시 데이터가 필요한 거 같아서 추가해뒀습니다.

green-yoon87 committed:
[feat] 필요한 유즈케이스 구현 ( 산책 요청 fetch, 산책 요청 응답)

green-yoon87 committed:
[fix] #66 유즈케이스 타입 클래스에서 구조체로 변경 및 파일이름 수정

green-yoon87 committed:
[feat] #66 Viewable 프로토콜 채택 후 구현

green-yoon87 committed:
[feat] #66 Presenter 구현

green-yoon87 committed:
[feat] #66 Interactor 구현

green-yoon87 committed:
[feat] Router 구현

pearhyunjin commented on SniffMeet/SniffMeet/Source/UseCase/RequestWalkUseCase.swift:
```diff
@@ -12,7 +12,7 @@ protocol RequestWalkUseCase {
 
 
-final class RequestWalkUseCaseImpl: RequestWalkUseCase {
+struct RequestWalkUseCaseImpl: RequestWalkUseCase {
```
이전에 지성님이 구현하신 UserCase는 전부 class 였는데 struct로 변경한 이유가 무엇인가요? (순수 궁금증입니다..)

green-yoon87 replied:
class보다 struct이 가진 장점이 커서(reference count 비용 없고 heap이 아닌 stack에 저장됨 ) 저는 될 수 있으면 struct을 사용하자는 주의입니다. 하나의 인스턴스를 공유하는 일이 없다고 판단하여 usecase가 클래스일 필요가 없어서 수정했습니당.

green-yoon87 committed:
[fix] #66 컴포넌트가 충분히 보이는 기기에서도 스크롤뷰가 적용되는 버그 고침

뷰의 크기와 650(se를 기준으로 보여야하는 최소 길이) 비교하여 max 값을 scrollview의 contentView의 height로 설정

kelly-chui reviewed (approved):
SE때문에 고생이 많으시네요
수고하셨습니다~👍

+ 추가적으로 영상에서 봤을 때 버튼을 지나칠 수도 있겠다라는 생각을 했는데 이 부분은 함께 고민해봐야할 것 같아요...

pearhyunjin commented on SniffMeet/SniffMeet/Source/Common/PaddingLabel.swift:
```diff
@@ -0,0 +1,32 @@
+
+import UIKit
+
+final class AllPaddingLabel: UILabel {
```
이름을 AllPaddingLabel로 하신 이유가 궁금합니다! TextView에서만 지정되는 padding 값 아닌가요~? 🤔

green-yoon87 replied:
요건 대면으로 답변드렸습니당 XD

pearhyunjin commented on SniffMeet/SniffMeet/Source/Walk/RequestWalk/View/ProfileView.swift:
```diff
@@ -0,0 +1,72 @@
+        view.contentMode = .scaleAspectFill
+        return view
+    }()
+    private var firstKeywordView: KeywordView = {
+        let view = KeywordView(title: "")
+        return view
+    }()
+    private var secondKeywordView: KeywordView = {
+        let view = KeywordView(title: "")
+        return view
+    }()
```
이후에 키워드 개수가 변경될수도 있으니, stackView를 사용해보는 건 어떨까요?

green-yoon87 replied:
헉 너무 좋네요! spacing도 통일할 수 있고 추가 작업할 때 이와 같이 수정하도록 하겠습니당

green-yoon87 committed:
[chore] #66 .pbxproj 파일 수정

pearhyunjin commented on SniffMeet/SniffMeet/Source/Walk/RespondWalk/Presenter/RespondWalkPresenter.swift:
```diff
@@ -0,0 +1,80 @@
+//
+//  Created by 윤지성 on 11/20/24.
+//
+import Foundation
+import Combine
```
순서 바꿔주세요~😁

green-yoon87 replied:
이번이 두번째네요.. 저의 영어 실력.... ㅠ

pearhyunjin reviewed (approved):
고생하셨습니다~!!

soletree reviewed (approved):
따봉~~~ 고생하셨습니다👍👍

green-yoon87 committed:
[fix] #66 사용하지 않는 프레임워크 임포트 제외

green-yoon87 committed:
Merge branch 'dev' into feature/#66/respond-walk-view
